[Windows] Drop support of older version of Windows.

basuke opened the pull request:
#### d27a9b023902534f01844dd91c56de14ee95f493
<pre>
[Windows] Drop support of older version of Windows.
<a href="https://bugs.webkit.org/show_bug.cgi?id=256314">https://bugs.webkit.org/show_bug.cgi?id=256314</a>

Reviewed by Don Olmstead and Yusuke Suzuki.

Change the supported Windows version from Windows 7 to Windows 10 1809 &quot;Redstone 5&quot;.
Version 1809 is the oldest version which is still supported in Enterprise LTSB/LTSC
editions.

<a href="https://learn.microsoft.com/en-us/windows/release-health/supported-versions-windows-client">https://learn.microsoft.com/en-us/windows/release-health/supported-versions-windows-client</a>

* Source/cmake/OptionsWin.cmake:

Canonical link: <a href="https://commits.webkit.org/263812@main">https://commits.webkit.org/263812@main</a>
</pre>

donny-dont reviewed (approved):
Are you sure we can drop the `WINVER` and `_WIN32_WINNT` declarations? If not looks like it should be set to `0x0A00`.

Constellation reviewed (approved):
r=me given that Windows 8. 8.1 are EOL (Jan 2023).

basuke committed:
[Windows] Drop support of older version of Windows.
https://bugs.webkit.org/show_bug.cgi?id=256314

Reviewed by Don Olmstead and Yusuke Suzuki.

Change the supported Windows version from Windows 7 to Windows 10 1809 "Redstone 5".
Version 1809 is the oldest version which is still supported in Enterprise LTSB/LTSC
editions.

https://learn.microsoft.com/en-us/windows/release-health/supported-versions-windows-client

* Source/cmake/OptionsWin.cmake:

Canonical link: https://commits.webkit.org/263812@main
